DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -108,29 +49,31 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
         /// <returns></returns>
         public async Task<DeviceModel> AddDeviceAsync(DeviceModel device)
         {
-            string deviceId = device.DeviceProperties.DeviceID;
-            DeviceModel existingDevice = await this.GetDeviceAsync(deviceId);
+            if (string.IsNullOrEmpty(device.id))
+            {
+                device.id = Guid.NewGuid().ToString();
+            }
 
+            var id = device.id;
+            DeviceModel existingDevice = await GetDeviceAsync(device.DeviceProperties.DeviceID);
```

Comment:
no null check on the DeviceID before its used. GetDeviceAsync has no handler for a null id